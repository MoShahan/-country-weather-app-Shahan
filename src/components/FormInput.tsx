import { FormInputProps } from "../types";
import "../styles/FormInput.css"

const FormInput = ({
    children,
    onChange,
    searchText,
    type = "text",
    id,
    handleSubmitClick
}: FormInputProps) => {

    return (
        <div className="searchBox">
            <form action="#" onSubmit={handleSubmitClick}>
                <label htmlFor={id}></label>
                <input
                    value={searchText}
                    id={id}
                    type={type}
                    onChange={onChange}
                    placeholder="Country Name"
                />
                
                <button type="submit"> Search </button>
            </form>
            {/* <a className="searchButton" href="#">
                    <i className="fas fa-search">
                    </i>
                </a> */}
        </div>
    );
};

export default FormInput;

// Submit button should be disabled if the input text is 
// empty / invalid (numbers, special characters, etc)