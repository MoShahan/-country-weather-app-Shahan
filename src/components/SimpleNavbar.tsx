import React from 'react'

function SimpleNavbar() {
  return (
    <div>SimpleNavbar</div>
  )
}

export default SimpleNavbar