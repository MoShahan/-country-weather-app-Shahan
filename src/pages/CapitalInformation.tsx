import React from 'react'
import { useParams } from 'react-router-dom'

const CAPITAL_API = "http://api.weatherstack.com/current?access_key=88be182651a79f2023dfd07c8192c86b&query="



function CapitalInformation() {

  const pathParams = useParams()

  console.log(pathParams.capitalCity)

  return (
    <div>Capital  Information</div>
  )
}

export default CapitalInformation