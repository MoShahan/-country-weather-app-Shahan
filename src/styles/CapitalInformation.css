.capitalBody {
    background-image: url("../images/weather-capital-bg.jpg");
    height: 100vh;
    background-size: cover;
    background-repeat: no-repeat;
}

.capitalBody>h1 {
    padding: 20px 10px;
    background-color: rgba(0, 0, 0, 0.801);
    color: rgb(243, 217, 182);
    margin: 0 275px;
    border-radius: 10px;
}

.capitalInfoTable {
    text-align: left;
    margin: auto;
    font-size: 1.1rem;
    margin-top: 100px;
}

.capitalInfoTable thead {
    font-weight: bolder;
    color: blueviolet;
}

.capitalInfoTable td {
    padding: 10px 70px;
}

.capitalInfoTable tbody td {
    color: white;
}

.capitalInfoTable tbody td:nth-child(odd) {
    color: rgb(46, 45, 45);
}

.capitalInfoTable tbody td:nth-child(even) {
    color: black;
    font-size: 1.1rem;
    font-weight: bold;
}