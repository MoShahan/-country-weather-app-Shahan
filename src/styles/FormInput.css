#searchInput {
    background-color: lightblue;
    padding: 10px;
    margin-left: 20px;
    border-radius: 25px;
    border: none;
}

.searchBox {
    background-color: rgba(64, 54, 211, 0.658);
    padding: 20px;
    margin-left: 400px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-radius: 14px;
}

.searchBox button {
    margin-left: 20px;
    border-radius: 5px;
    background-color: rgba(181, 181, 240, 0.39);
    transition: all 0.3s;
}

.searchBox button:hover {
    cursor: pointer;
    transform: scale(1.3);
    background-color: #9d9dee;
}