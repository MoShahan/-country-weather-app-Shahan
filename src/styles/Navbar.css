.navbar {
    background-color: red;
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 75%;
    margin: auto;
    padding: 10px;
}

.nav-heading {
    display: flex;
}

.weather-icon {
    height: 100px;
}