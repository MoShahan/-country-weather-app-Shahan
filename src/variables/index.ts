export const appTitle = "COUNTRY WEATHER APP"

export const COUNTRY_API = "https://restcountries.com/v3.1/name/"

const ACCESS_KEY = "076f152151d94ca04e96d05d13042e24"

export const CAPITAL_API = `http://api.weatherstack.com/current?access_key=${ACCESS_KEY}&query=`