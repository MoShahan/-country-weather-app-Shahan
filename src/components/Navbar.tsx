import React from 'react'
import { NavbarType } from '../types';
import FormInput from './FormInput';
import "../styles/Navbar.css"
import image from "../images/weather-app-icon.png"

export default function Navbar({ title, searchText, handleChange }: NavbarType) {

    console.log("Navbar Called")

    return (
        <nav className='navbar'>
            <div className="nav-heading">
                <img className="weather-icon" src={image} />
                <h1>{title}</h1>
            </div>
            <FormInput
                searchText={searchText}
                onChange={handleChange}
                id="searchInput"
            >
                Search
            </FormInput>
        </nav>
    )
}
