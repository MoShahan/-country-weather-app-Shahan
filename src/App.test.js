import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import FormInput from "./components/FormInput";
import Navbar from "./components/Navbar";
import SimpleNavbar from "./components/SimpleNavbar";
import CapitalInformation from "./pages/CapitalInformation";
import { appTitle } from "./variables";

describe("Basic test", () => {
    test("Checking if the Title is WEATHER APP", () => {
        expect(appTitle).toBe("WEATHER APP");
    });
});



// unit testing

describe("testing Simple Navbar component", () => {
    test("rendering", () => { 
        render(
            <SimpleNavbar/>
        )
    })
})

describe("testing FormInput component", () => {
    test("rendering FormInput component", () => {
        const FormInputProps = {
            onChange: jest.fn(),
            searchText: "India",
        };
        render(
            <FormInput {...FormInputProps} />
        );
    });

    test("checking if onChange function is called", () => {
        const FormInputProps = {
            onChange: jest.fn(),
            searchText: "India",
        };
        render(
            <FormInput {...FormInputProps} />
        );
        const inputBox = screen.getByDisplayValue("India");
        // screen.debug();
        fireEvent.change(inputBox, { target: { value: "United" } });
        fireEvent.change(inputBox, { target: { value: "UAE" } });
        expect(FormInputProps.onChange).toHaveBeenCalledTimes(2);
    });
})



// integration testing

describe("Rendering the Capital component", () => {
    test("checking if render works", () => {
        render(
            <BrowserRouter>
                <CapitalInformation />
            </BrowserRouter>
        );
    });
});



// snapshot

describe("SNAPSHOT", () => {
    test("snapshot testing", () => {
        const NavbarProps = {
            title: "TEST",
            searchText: "Country"
        };
        const { NavMain } = render(
            < Navbar {...NavbarProps} />
        );
        expect(NavMain).toMatchSnapshot();
    });
})



// Your code must have at least 80 % code coverage, which
// means all your code must have unit, integration and
// snapshot test.You can check the coverage by running the
// command: npm test--coverage