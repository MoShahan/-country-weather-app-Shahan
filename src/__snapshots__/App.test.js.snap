// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOT snapshot testing 1`] = `undefined`;
